src/Data/Macaw/CFGSlice.hs

```diff
@@ -126,15 +128,25 @@ assignBlockEnd archFns blendVar stmt = MSB.crucGenArchConstraints archFns $ do
   blend' <- blockEndAtom archFns blend
   MSB.addStmt $ CR.WriteGlobal blendVar blend'
 
+blockEndCaseEq :: IsSymInterface sym
+               => Proxy arch
```

Comment:
It is generally a best practice to make proxies in positive (argument) positions type variables:

```
blockEndCaseEq :: forall sym arch proxy => IsSymInterface sym -> proxy arch -> ...
```

This way the caller can pass in any valid value of the correct kind (given that this function doesn't care what is passed in).